KaitaiStream.ts

```diff
@@ -0,0 +1,850 @@
+      Object.setPrototypeOf(this, KaitaiStream.ValidationLessThanError.prototype);
+      this.name = "ValidationLessThanError";
+      this.message = "not in range, min [" + min + "], but got [" + actual + "]";
+      this.min = min;
+      this.actual = actual;
+    }
+  };
+
+  static ValidationGreaterThanError = class extends Error {
+    max: any;
+    actual: any;
+
+    constructor(max, actual) {
+      super();
+      // Workaround https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+      Object.setPrototypeOf(this, KaitaiStream.ValidationGreaterThanError.prototype);
+      this.name = "ValidationGreaterThanError";
+      this.message = "not in range, max [" + max + "], but got [" + actual + "]";
+      this.max = max;
+      this.actual = actual;
+    }
+  };
+
+  static ValidationNotAnyOfError = class extends Error {
+    actual: any;
+
+    constructor(actual, io, srcPath) {
+      super();
+      // Workaround https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+      Object.setPrototypeOf(this, KaitaiStream.ValidationNotAnyOfError.prototype);
+      this.name = "ValidationNotAnyOfError";
+      this.message = "not any of the list, got [" + actual + "]";
+      this.actual = actual;
+    }
+  };
+
+  /**
+    Ensures that we have an least `length` bytes left in the stream.
+    If that's not true, throws an EOFError.
+
+    @param {number} length Number of bytes to require
+    */
+  ensureBytesLeft(length) {
+    if (this.pos + length > this.size) {
+      throw new KaitaiStream.EOFError(length, this.size - this.pos);
+    }
+  };
+
+  /**
+    Maps a Uint8Array into the KaitaiStream buffer.
+  
+    Nice for quickly reading in data.
+  
+    @param {number} length Number of elements to map.
+    @return {Object} Uint8Array to the KaitaiStream backing buffer.
+    */
+  mapUint8Array(length) {
+    length |= 0;
+
+    this.ensureBytesLeft(length);
+
+    var arr = new Uint8Array(this._buffer, this.byteOffset + this.pos, length);
+    this.pos += length;
+    return arr;
+  };
+
+  /**
+    Creates an array from an array of character codes.
+    Uses String.fromCharCode in chunks for memory efficiency and then concatenates
+    the resulting string chunks.
+  
+    @param {array|Uint8Array} array Array of character codes.
+    @return {string} String created from the character codes.
+    */
+  static createStringFromArray = function (array) {
+    var chunk_size = 0x8000;
+    var chunks = [];
+    var useSubarray = typeof array.subarray === 'function';
+    for (var i = 0; i < array.length; i += chunk_size) {
+      chunks.push(String.fromCharCode.apply(null, useSubarray ? array.subarray(i, i + chunk_size) : array.slice(i, i + chunk_size)));
+    }
+    return chunks.join("");
+  };
+}
+
+export default KaitaiStream;
```

Comment:
Missing EOF newline.

Reply:
just pushed an udpated commit which fixes this, here and in rollup.config.js